### ISSUE TITLE
Rails shouldn't enforce the use of secrets.yml

### ISSUE THREAD

**Issue by krainboltgreene** [rails/rails#20390]
```
`validate_secret_key_config!': Missing `secret_token` and `secret_key_base` for 'development' environment, set these values in `config/secrets.yml`
```

It should be sufficient that these ENV values exist and contain something. That they exist some arbitrary file Rails decided on seems to stomp all over other tools.

It's also (as far as I can tell) Rails has cared *where* something is defined.

**Comment by rafaelfranca** [comment]
Sorry but we centralised the lookup of these values on that file, so you can use environment variables, a file vault or wherever you want to use to store your secrets. The file have to be there.

**Comment by krainboltgreene** [comment]
I get why you do it, I'm not an idiot.

But why not just expect the values to be in the ENV? Wouldn't that handle
both situations?

On Mon, Jun 1, 2015 at 10:30 AM, Rafael Mendonça França <
notifications@github.com> wrote:

> Closed #20390 <https://github.com/rails/rails/issues/20390>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/issues/20390#event-319020377>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115

**Comment by rafaelfranca** [comment]
Just to not have to maintain a special case on the code. The environment
variable case is already handled on the file case so I don't see why
handling it using a different path

**Comment by krainboltgreene** [comment]
In the code make one check:

`if ENV["SECRET_SESSION_KEY"]`

BAM done, now you support both ways and no one is forced to keep a file
around.

On Mon, Jun 1, 2015 at 2:19 PM, Rafael Mendonça França <
notifications@github.com> wrote:

> Just to not have to maintain a special case on the code. The environment
> variable case is already handled on the file case so I don't see why
> handling it using a different path
>
> On Mon, Jun 1, 2015, 16:15 Kurtis Rainbolt-Greene <
> notifications@github.com>
> wrote:
>
> > I get why you do it, I'm not an idiot.
> >
> > But why not just expect the values to be in the ENV? Wouldn't that handle
> > both situations?
> >
> > On Mon, Jun 1, 2015 at 10:30 AM, Rafael Mendonça França <
> > notifications@github.com> wrote:
> >
> > > Closed #20390 <https://github.com/rails/rails/issues/20390>.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/rails/rails/issues/20390#event-319020377>.
> > >
> >
> >
> >
> > --
> > Kurtis Rainbolt-Greene, Hacker
> > Software Developer
> > 1631 8th St.
> > New Orleans, LA
> > 70115
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/rails/rails/issues/20390#issuecomment-107675035>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/issues/20390#issuecomment-107675755>.
>



-- 
Kurtis Rainbolt-Greene, Hacker
Software Developer
1631 8th St.
New Orleans, LA
70115
